enabler/src/de/schildbach/pte/VrsProvider.java

```diff
@@ -0,0 +1,861 @@
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.TimeZone;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.annotation.Nullable;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import com.google.common.base.Charsets;
+
+import de.schildbach.pte.dto.Departure;
+import de.schildbach.pte.dto.Fare;
+import de.schildbach.pte.dto.Line;
+import de.schildbach.pte.dto.LineDestination;
+import de.schildbach.pte.dto.Location;
+import de.schildbach.pte.dto.LocationType;
+import de.schildbach.pte.dto.NearbyLocationsResult;
+import de.schildbach.pte.dto.Point;
+import de.schildbach.pte.dto.Position;
+import de.schildbach.pte.dto.Product;
+import de.schildbach.pte.dto.QueryDeparturesResult;
+import de.schildbach.pte.dto.QueryTripsContext;
+import de.schildbach.pte.dto.QueryTripsResult;
+import de.schildbach.pte.dto.ResultHeader;
+import de.schildbach.pte.dto.StationDepartures;
+import de.schildbach.pte.dto.Stop;
+import de.schildbach.pte.dto.Style;
+import de.schildbach.pte.dto.SuggestLocationsResult;
+import de.schildbach.pte.dto.SuggestedLocation;
+import de.schildbach.pte.dto.Trip;
+import de.schildbach.pte.dto.Trip.Leg;
+import de.schildbach.pte.util.ParserUtils;
+
+/**
+ * @author Michael Dyrna
+ */
+public class VrsProvider extends AbstractNetworkProvider {
+	private static class Context implements QueryTripsContext {
+		private static final long serialVersionUID = 8758709354176420641L;
+		private Date lastDeparture = null;
+		private Date firstArrival = null;
+		public Location from;
+		public Location via;
+		public Location to;
+		public Set<Product> products;
+
+		private Context() {
+		}
+
+		public boolean canQueryLater() {
+			return true;
+		}
+
+		public boolean canQueryEarlier() {
+			return true;
+		}
+
+		public void departure(Date departure) {
+			if (this.lastDeparture == null || this.lastDeparture.compareTo(departure) < 0) {
+				this.lastDeparture = departure;
+			}
+		}
+
+		public void arrival(Date arrival) {
+			if (this.firstArrival == null || this.firstArrival.compareTo(arrival) > 0) {
+				this.firstArrival = arrival;
+			}
+		}
+
+		public Date getLastDeparture() {
+			return lastDeparture;
+		}
+
+		public Date getFirstArrival() {
+			return firstArrival;
+		}
+	}
+
+	// valid host names: www.vrsinfo.de, android.vrsinfo.de, ios.vrsinfo.de
+	// performance comparison March 2015 showed www.vrsinfo.de to be fastest for trips
+	protected static final String API_BASE = "http://www.vrsinfo.de/index.php";
+	protected static final boolean httpPost = false;
+	protected static final String SERVER_PRODUCT = "vrs";
+	protected static final SimpleDateFormat dateTimeFormat = new SimpleDateFormat("yyyy-MM-dd'T'kk:mm:ssZ");
+	protected static final Pattern nameWithPosition = Pattern.compile("(.*) - (.*)");
+
+	protected static final Map<String, Style> STYLES = new HashMap<String, Style>();
+
+	static
+	{
+		// Schnellbusse VRR
```

Comment:
Out of curiosity: where have you got these colors from? Are you aware VrrProvider currently also defines colors for VRS lines? Are there any differences?